api/tests.py

```diff
@@ -71,3 +80,274 @@ def test_options_manifest_data(self):
+            "group": "Bad Guys",
+            "subtype": "goblinoid",
+            "alignment": "neutral evil",
+            "armor_class": 15,
+            "hit_points": 7,
+            "hit_dice": "2d6",
+            "speed": "30 ft.",
+            "strength": 8,
+            "dexterity": 14,
+            "constitution": 10,
+            "intelligence": 10,
+            "wisdom": 8,
+            "charisma": 8,
+            "stealth": 6,
+            "damage_vulnerabilities": "",
+            "damage_resistances": "",
+            "damage_immunities": "",
+            "condition_immunities": "",
+            "senses": "darkvision 60 ft., passive Perception 9",
+            "languages": "Common, Goblin",
+            "challenge_rating": "1/4",
+            "special_abilities": [
+                {
+                    "name": "Nimble Escape",
+                    "desc": "The goblin can take the Disengage or Hide action as a bonus action on each of its turns.",
+                    "attack_bonus": 0
+                }
+            ],
+            "actions": [
+                {
+                    "name": "Scimitar",
+                    "desc": "Melee Weapon Attack: +4 to hit, reach 5 ft., one target. Hit: 5 (1d6 + 2) slashing damage.",
+                    "attack_bonus": 4,
+                    "damage_dice": "1d6",
+                    "damage_bonus": 2
+                },
+                {
+                    "name": "Shortbow",
+                    "desc": "Ranged Weapon Attack: +4 to hit, range 80/320 ft., one target. Hit: 5 (1d6 + 2) piercing damage.",
+                    "attack_bonus": 4,
+                    "damage_dice": "1d6",
+                    "damage_bonus": 2
+                }
+            ],
+            "speed_json": {
+                "walk": 30
+            },
+            "armor_desc": "leather armor, shield",
+            "page_no": 315
+        }
+        """
+        i = Importer(ImportOptions(update=True, append=False, testrun=False))
+        i.import_document(json.loads(self.test_document_json),ImportSpec("test_filename","test_model_class","import_monster"))
+        i.import_monster(json.loads(self.test_monster_json),ImportSpec("test_filename","test_model_class","import_monster"))
+
+    def test_get_monsters(self):
+        response = self.client.get(f'/monsters/?format=json')
+        # Confirm the basic elements show up.
+        self.assertContains(response,'count',count=1)
+        self.assertContains(response,'next',count=1)
+        self.assertContains(response,'previous',count=1)
+        self.assertContains(response,'results',count=1)        
+
+    def test_get_monster_data(self):
+        response = self.client.get(f'/monsters/?format=json')        
+        self.assertEqual(response.json()['count'],1)
+        in_goblin = json.loads(self.test_monster_json)
+        out_goblin = response.json()['results'][0]
+
+        self.assertEqual(in_goblin['name'], out_goblin['name'])
```

Comment:
I think it would be much better to consolidate this logic, like:

```py
equal_fields = ('name', 'size', 'type', 'subtype', 'group', 'alignment', ...)
for field_name in equal_fields:
    self.assertEqual(in_goblin[field_name], out_goblin[field_name])

# Some field names unfortunately cannot be identical.
self.assertEqual(in_goblin['speed_json'], out_goblin['speed']

# Some fields are expected to be None
none_fields = ('strength_save', 'dexterity_save', ...)
for field_name in none_fields:
    self.assertNone(out_goblin[field_name])
# then do the same thing for empty strings, too
```